interop/xds/client/client.go

```diff
@@ -368,6 +378,22 @@ func parseRPCMetadata(rpcMetadataStr string, rpcs []string) []*rpcConfig {
 
 func main() {
 	flag.Parse()
+	if *enableCSMObservability {
+		exporter, err := prometheus.New()
+		if err != nil {
+			logger.Fatalf("Failed to start prometheus exporter: %v", err)
+		}
+		provider := metric.NewMeterProvider(
+			metric.WithReader(exporter),
+		)
+		go http.ListenAndServe("0.0.0.0:9464", promhttp.Handler())
```

Comment:
This is a defined number from the testing design: "opentelemetry::exporter::metrics::PrometheusExporterOptions opts;
// default was "localhost:9464" which causes connection issue
// across GKE pods
opts.url = "0.0.0.0:9464";" 

"When the client / server GKE pods are running, the Prometheus metrics should be available at the port :9464."

If this port is in use the test will fail. This is unexpected, as I think the tests run synchronously with defined client/server image behavior so nothing will collide. If the port collides, that will be an infra issue I think.

Reply:
The 0s don't seem correct to me, though, because that limits us to IPv4?  I think you want `:9464` to attach the listener onto all networks.  But also this seems like something that should be a flag?

Reply:
"Stanley Cheung, Now
Sergii and I talked about this and we are fine with leaving this hard coded for now. We are aware of this and if there's a need we can pass that as a flag. Just don't see a need right now"

Reply:
"Stanley Cheung, Now
I somehow ran into a problem for the c++ exporter that the default URL doesn't work and I have to override it with [0.0.0.0:9464](http://0.0.0.0:9464/) even though the default port is 9464"

Reply:
"Let's do that in Go too. If there's a problem then at least we know all of the lang clients have the same thing to change"